crates/amf0/src/value.rs

```diff
@@ -280,12 +280,120 @@ impl serde::ser::Serialize for Amf0Value<'_> {
     }
 }
 
+#[cfg(feature = "serde")]
+#[cfg_attr(docsrs, doc(cfg(feature = "serde")))]
+impl<'de, 'a: 'de> serde::Deserializer<'de> for &'a Amf0Value<'de> {
+    type Error = Amf0Error;
+
+    serde::forward_to_deserialize_any! {
+        // need to handle these number types from i8 thru f64; try to deserialze them as their original type then forward as f64
+        bool i8 i16 i32 i64 u8 u16 u32 u64 f32 f64 char str string unit
```

Comment:
Here where we have `i8 i16 i32 i64 u8 u16 u32 u64 f32` we should convert try convert f64 to these types if it fits in that type (see `num_traits` crate)